os/net/ipv6/uip-ds6-nbr.c

```diff
@@ -125,19 +264,52 @@ uip_ds6_nbr_rm(uip_ds6_nbr_t *nbr)
     return nbr_table_remove(ds6_neighbors, nbr);
   }
   return 0;
+#endif /* UIP_DS6_NBR_MULTI_IPV6_ADDRS */
 }
 
 /*---------------------------------------------------------------------------*/
 int
 uip_ds6_nbr_update_ll(uip_ds6_nbr_t **nbr_pp, const uip_lladdr_t *new_ll_addr)
 {
+#if UIP_DS6_NBR_MULTI_IPV6_ADDRS
+  uip_ds6_nbr_entry_t *nbr_entry;
+  uip_ds6_nbr_t *nbr;
+#else
   uip_ds6_nbr_t nbr_backup;
+#endif /* UIP_DS6_NBR_MULTI_IPV6_ADDRS */
 
   if(nbr_pp == NULL || new_ll_addr == NULL) {
     LOG_ERR("%s: invalid argument\n", __func__);
     return -1;
   }
 
+#if UIP_DS6_NBR_MULTI_IPV6_ADDRS
```

Comment:
The exact behavior of this function with/without `UIP_DS6_NBR_MULTI_IPV6_ADDRS` will need some commenting and ideally Doxygen.